vayesta/solver/coupling.py

```diff
@@ -427,16 +569,39 @@ def externally_correct(solver, external_corrections):
         dt2y = transform_amplitude(dt2y, rxy_occ, rxy_vir, inverse=True)
         dt1 = spinalg.add(dt1, dt1y)
         dt2 = spinalg.add(dt2, dt2y)
+
+        if corrtype == 'external-ccsdv':
+            # Include the t3v term, contracting with the integrals from the x cluster
+            # These have already been fragment projected, and rotated into the x cluster
+            # in this function.
+            dt2y_t3v = _get_delta_t2_from_t3v(gvvov_x, gooov_x, govoo_x, fy, rxy_occ, rxy_vir, cxs_occ, projectors)
+            dt2 = spinalg.add(dt2, dt2y_t3v)
+
+        # Note that the lines below don't actually give the norm of the T updates, since they are missing the
+        # energy denominators
         solver.log.info("External correction from fragment %3d (%s):  dT1= %.3e  dT2= %.3e",
                         fy.id, fy.solver, *get_amplitude_norm(dt1y, dt2y))
+    solver.log.info("Total external correction from all fragments:  dT1= %.3e  dT2= %.3e",*get_amplitude_norm(dt1, dt2))
 
     if solver.spinsym == 'restricted':
 
         def callback(kwargs):
             """Add external correction to T1 and T2 amplitudes."""
             t1, t2 = kwargs['t1new'], kwargs['t2new']
-            t1[:] += dt1
-            t2[:] += dt2
+
+            # Note that this will not work with a level shift
+            # TODO: Do this properly, using the correct MO energies, and precontracting
+            # outside the callback function.
+            eris = kwargs['eris']
+            mo_energy = eris.mo_energy
+            nocc = t1.shape[0]
+            eia = mo_energy[:nocc,None] - mo_energy[None,nocc:]
+            eijab = mo_energy[:nocc,None,None,None] + mo_energy[None,:nocc,None,None] \
+                    - mo_energy[None,None,nocc:,None] - mo_energy[None,None,None,nocc:]
+
+            # Divide by energy denominators
+            t1[:] += (dt1 / eia)
+            t2[:] += (dt2 / eijab)
```

Comment:
This breaks the `delta-tailor` approach, or not?

Reply:
Yes - fixed in latest push.